### PR TITLE
Change logic fav security

### PR DESCRIPTION
**Pull request by remy-rm**
corrige bug and adapt for match frontEnd

### CODE REVIEW

**Review by td6904 — APPROVED** [review]

File: src/main/java/com/templateproject/api/controller/FavoriteController.java [hunk 1]
```diff
@@ -16,48 +16,56 @@
...
-            return ResponseEntity.badRequest().body("");
+            return ResponseEntity.badRequest().body("Invalid user ID or recipe ID");
+        }
+        Users user = optionalUser.get();
+        Recipe recipe = optionalRecipe.get();
+
+        Optional<FavoriteRecipes> existingFavoriteRecipe = favoriteRecipeRepository.findByUsersAndRecipe(user, recipe);
+
+        if (existingFavoriteRecipe.isPresent()) {
+            // Si la relation existe, la supprimer
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/controller/FavoriteController.java]: remove dead code

File: src/main/java/com/templateproject/api/controller/FavoriteController.java [hunk 2]
```diff
@@ -16,48 +16,56 @@
...
+            favoriteRecipeRepository.delete(existingFavoriteRecipe.get());
+            return ResponseEntity.ok("Favorite recipe removed");
+        } else {
+            // Si la relation n'existe pas, la créer
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/controller/FavoriteController.java]: remove dead code

File: src/main/java/com/templateproject/api/controller/FavoriteController.java [hunk 3]
```diff
@@ -16,48 +16,56 @@
...
 
 
 
     @DeleteMapping("/delete")
     public ResponseEntity<String> deleteFavoriteRecipe(@RequestParam long recipeId, @RequestParam long userId) {
-        Optional<FavoriteRecipes> optionalFavoriteRecipe = favoriteRecipeRepository.findByRecipeIdAndUserId(recipeId, userId);
+        Optional<FavoriteRecipes> optionalFavoriteRecipe = favoriteRecipeRepository.findByRecipeIdAndUsersId(recipeId, userId);
 
         if (optionalFavoriteRecipe.isEmpty()) {
             return ResponseEntity.badRequest().body("Favorite_recipe introuvable pour l'utilisateur donné et l'ID de recette donné.");
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/controller/FavoriteController.java]: why using french while your whole application is in english ?

File: src/main/java/com/templateproject/api/service/UsersService.java [hunk 4]
```diff
@@ -16,62 +20,70 @@
...
-                new UsernamePasswordAuthenticationToken(email, password)
-        );
-        return tokenService.generateToken(authentication);
+        try {
+            Authentication authentication = this.authManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(email, password)
+            );
+            return tokenService.generateToken(authentication);
+        } catch (Exception e) {
+            // Log the error or exception details here
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/service/UsersService.java]: remove dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**